tools/mc-java/src/main/java/io/spine/tools/mc/java/codegen/SignalConfig.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.mc.java.codegen;
+
+import com.google.common.collect.ImmutableSet;
+import io.spine.base.MessageFile;
+import io.spine.base.SignalMessage;
+import io.spine.tools.protoc.FilePattern;
+import io.spine.tools.protoc.ForSignals;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.gradle.api.Project;
+
+/**
+ * Code generation config for a type of signal messages.
+ *
+ * <p>May configure all the events, all the rejections, or all the commands.
+ *
+ * <p>The configuration applied to events does not automatically apply to rejections.
+ */
+public final class SignalConfig extends MessageGroupConfig<ForSignals> {
+
+    SignalConfig(Project p) {
+        super(p);
+    }
+
+    void convention(MessageFile file, Class<? extends SignalMessage> interfaceClass) {
```

Comment:
These undocumented methods are strangely package-private. Which probably means they are made for the whole package to use those.

Please consider documenting those.